ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala

```diff
@@ -88,9 +88,7 @@ private[kvutils] object InputsAndEffects {
                 GlobalKey(create.coinst.template, forceNoContractIds(keyWithMaintainers.key.value)))
             }
 
-          case exe @ NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _) =>
-            addContractInput(exe.targetCoid)
-
+          case NodeExercises(_, _, _, _, _, _, _, _, _, _, _, _, _, _) =>
```

Comment:
So many underscores. I'd find this whole block easier to read if we used this structure:

```suggestion
          case _: NodeExercises =>
```

Reply:
NodeExercise is a parametric type. I really do not like pattern matching using `parametric type`

Reply:
```suggestion
          case NodeExercises[_, _, _, _, _, _, _, _, _, _, _, _, _, _] =>
```
:stuck_out_tongue: